db/collection_get.go

```diff
@@ -48,13 +48,19 @@ func (c *collection) get(
 	ctx context.Context,
 	txn datastore.Txn,
 	key core.PrimaryDataStoreKey,
+	fields []*client.FieldDescription,
```

Comment:
I didn't realise `fetcher` didn't use the `fields` because I didn't dig it the much. But it might change in the future and I'd say it should. Anyway I think that API for `Fetcher` needs to change first.